### DISCUSSION TITLE
Why is the default InstanceContextMode is "PerSession" ?

### DISCUSSION THREAD

**Discussion by nicoclau** [CoreWCF/CoreWCF#1484] [Q&A]
Hello all I noticed that the default InstanceContextMode is "PerSession", before it was "PerCall".

Why was that changed?

Thanks

**Comment by mconnew** [comment]
[Here's the code](https://github.com/microsoft/referencesource/blob/master/System.ServiceModel/System/ServiceModel/ServiceBehaviorAttribute.cs#L176) for .NET Framework where `ServiceBehaviorAttribute.InstanceContextMode` is defined. It's attributed with a default value, which you can see is `PerSesssion`. The `DefaultValue` attribute is only used when reading/writing from config, so this in itself doesn't show that's the default, it's just an indicator, but if you look through the code, nowhere is `InstanceContextMode` set other than through the property, there isn't even a default value configured in the constructor. This is because the default value of an enum is the 0'th value. If you look at [the enum definition](https://github.com/microsoft/referencesource/blob/master/System.ServiceModel/System/ServiceModel/InstanceContextMode.cs#L7-L12), you can see the first value is `PerSession`, making it the default value. As you can see, the default `InstanceContextMode` on .NET Framework is `PerSession`.  

If we look at [the CoreWCF implementation](https://github.com/CoreWCF/CoreWCF/blob/v0.1.0/src/CoreWCF.Primitives/src/CoreWCF/ServiceBehaviorAttribute.cs#L95-L108), the very first release (v0.1.0), you can see the default was also `PerSession`.  

What you might have experienced was `PerSession` behaves like `PerCall` when you use a sessionless binding such as BasicHttpBinding. This is because every call behaves like it's a one off session so every call gets a new instance. But when you switch to a binding with sessions (e.g. NetTcpBinding), then the session extends beyond the single call and lives (in the case of NetTcp) for the duration of the connection.  

What behavior did you experience that lead to you believing this had changed?
